src/function3d/smooth_periodic_function.hpp

```diff
@@ -27,13 +27,31 @@
 #include "SDDK/gvec.hpp"
 #include "utils/utils.hpp"
 #include "utils/profiler.hpp"
+#include "SDDK/type_definition.hpp"
 
 #ifndef __SMOOTH_PERIODIC_FUNCTION_HPP__
 #define __SMOOTH_PERIODIC_FUNCTION_HPP__
 
 namespace sirius {
 
-using double_complex = std::complex<double>;
+// type traits to handle Spfft driver for different precision type
+template <typename T>
+struct SPFFT_driver {};
```

Comment:
This should be moved to a more suitable location, similar to the real / complex type definition header, since this might be require elsewhere as well.

Reply:
there is SDDK/fft.hpp file to interface with SpFFT. It should go there